core/src/telemetry/otel.rs

```diff
@@ -120,6 +120,9 @@ pub(super) fn augment_meter_provider_with_defaults(
 pub fn build_otlp_metric_exporter(
     opts: OtelCollectorOptions,
 ) -> Result<CoreOtelMeter, anyhow::Error> {
+    global::set_error_handler(|err| {
+        tracing::error!("{}", err);
+    })?;
```

Comment:
I wonder if there is a better more global place to set this instead of on each `build_otlp_metric_exporter` call? Unsure if we have global metric initialization code.